lib/utils/utf8.c

```diff
@@ -79,3 +80,45 @@ char *utf8_lcpy(char *dst, const char *src, size_t n)
 
 	return dst;
 }
+
+int utf8_count_chars(const char *s)
+{
+	int count = 0;
+	const unsigned char *p = (const unsigned char *)s;
```

Comment:
Is the cast from `const char *s` to `const unsigned char *` necessary? 

Keep also in mind that values such as `0x80` are signed, so if you want to handle it strictly as unsigned, you probably should use `0x80U` (minor detail, can be skipped)